### PR TITLE
Use lazy range in gdb pretty printers

### PR DESCRIPTION
**Pull request by dzamlo**

### CODE REVIEW

File: src/etc/gdb_rust_pretty_printing.py [hunk 1]
```diff
@@ -10,8 +10,12 @@
 
 import gdb
 import re
+import sys
 import debugger_pretty_printers_common as rustpp
 
+if sys.version_info.major >= 3:
+    xrange = range
```
Comment (alexcrichton) [inline src/etc/gdb_rust_pretty_printing.py]: Could you be sure to add a comment here as well indicating why we're doing this?

**Comment by alexcrichton** [comment]
Ah I've gone ahead and added a comment in https://github.com/rust-lang/rust/pull/34689, and I'll try to land this as part of that.

**Comment by dzamlo** [comment]
Sorry, I didn't see your comment about adding a comment.

**Comment by alexcrichton** [comment]
No worries! Just a minor nit anyway :)

**Pull request merged**